app/app.go

```diff
@@ -459,14 +458,13 @@ func NewDesmosApp(
 		ibcfee.NewAppModule(app.IBCFeeKeeper),
 		ica.NewAppModule(&app.ICAControllerKeeper, &app.ICAHostKeeper),
 		wasm.NewAppModule(app.appCodec, &app.WasmKeeper, app.StakingKeeper, app.AccountKeeper, app.BankKeeper, app.MsgServiceRouter(), app.GetSubspace(wasmtypes.ModuleName)),
-	}
 
-	app.RegisterModules(legacyModules...)
-	cfg := app.Configurator()
-	for _, m := range legacyModules {
-		if s, ok := m.(module.HasServices); ok {
-			s.RegisterServices(cfg)
-		}
+		// register additional types to codec
+		ibctm.AppModuleBasic{},
+		solomachine.AppModuleBasic{},
+	}
```

Comment:
`RegisterModules` also covered `RegisterInterfaces`, so we don't need to handle it manually.